drivers/charger/charger_pf1550.c

```diff
@@ -0,0 +1,695 @@
+/*
+ * Copyright 2024 Arduino SA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_pf1550_charger
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/charger.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/linear_range.h>
+
+#include "zephyr/logging/log.h"
+LOG_MODULE_REGISTER(pf1550_charger);
+
+
+#define INT_ENABLE_DELAY K_MSEC(500)
+
+enum {
+  /* Charger Register Addresses */
+  CHARGER_CHG_INT           = 0x80 + 0x00,
+  CHARGER_CHG_INT_MASK      = 0x80 + 0x02,
+  CHARGER_CHG_INT_OK        = 0x80 + 0x04,
+  CHARGER_VBUS_SNS          = 0x80 + 0x06,
+  CHARGER_CHG_SNS           = 0x80 + 0x07,
+  CHARGER_BATT_SNS          = 0x80 + 0x08,
+  CHARGER_CHG_OPER          = 0x80 + 0x09,
+  CHARGER_CHG_TMR           = 0x80 + 0x0A,
+  CHARGER_CHG_EOC_CNFG      = 0x80 + 0x0D,
+  CHARGER_CHG_CURR_CFG      = 0x80 + 0x0E,
+  CHARGER_BATT_REG          = 0x80 + 0x0F,
+  CHARGER_BATFET_CNFG       = 0x80 + 0x11,
+  CHARGER_THM_REG_CNFG      = 0x80 + 0x12,
+  CHARGER_VBUS_INLIM_CNFG   = 0x80 + 0x14,
+  CHARGER_VBUS_LIN_DPM      = 0x80 + 0x15,
+  CHARGER_USB_PHY_LDO_CNFG  = 0x80 + 0x16,
+  CHARGER_DBNC_DELAY_TIME   = 0x80 + 0x18,
+  CHARGER_CHG_INT_CNFG      = 0x80 + 0x19,
+  CHARGER_THM_ADJ_SETTING   = 0x80 + 0x1A,
+  CHARGER_VBUS2SYS_CNFG     = 0x80 + 0x1B,
+  CHARGER_LED_PWM           = 0x80 + 0x1C,
+  CHARGER_FAULT_BATFET_CNFG = 0x80 + 0x1D,
+  CHARGER_LED_CNFG          = 0x80 + 0x1E,
+  CHARGER_CHGR_KEY2         = 0x80 + 0x1F,
+};
```

Comment:
usually in zephyr code register offets are in `#defines` rather than an enum, it's not like you are using the enum as an argument or in a switch anyway